### ISSUE TITLE
[gstreamer] gstDecoder -- failed to retrieve next image buffer

### ISSUE THREAD

**Issue by muttakigenc** [dusty-nv/jetson-inference#1383]
./detectnet --model=models/iha/best.onnx --labels=models/iha/labels.txt --input-blob=images --output-cvg=output --output-bbox=images
[gstreamer] initialized gstreamer, version 1.14.5.0
[gstreamer] gstCamera -- attempting to create device csi://0
[gstreamer] gstCamera pipeline string:
[gstreamer] nvarguscamerasrc sensor-id=0 ! video/x-raw(memory:NVMM), width=(int)1280, height=(int)720, framerate=30/1, format=(string)NV12 ! nvvidconv flip-method=2 ! video/x-raw(memory:NVMM) ! appsink name=mysink
[gstreamer] gstCamera successfully created device csi://0
[video]  created gstCamera from csi://0
------------------------------------------------
gstCamera video options:
------------------------------------------------
  -- URI: csi://0
     - protocol:  csi
     - location:  0
  -- deviceType: csi
  -- ioType:     input
  -- codec:      raw
  -- width:      1280
  -- height:     720
  -- frameRate:  30.000000
  -- bitRate:    0
  -- numBuffers: 4
  -- zeroCopy:   true
  -- flipMethod: rotate-180
  -- loop:       0
  -- rtspLatency 2000
------------------------------------------------
[OpenGL] failed to open X11 server connection.
[OpenGL] failed to create X11 Window.
detectnet:  failed to create output stream

detectNet -- loading detection network model from:
          -- prototxt     NULL
          -- model        models/iha/best.onnx
          -- input_blob   'images'
          -- output_cvg   'output'
          -- output_bbox  'images'
          -- mean_pixel   0.000000
          -- mean_binary  NULL
          -- class_labels models/iha/labels.txt
          -- threshold    0.500000
          -- batch_size   1

[TRT]    TensorRT version 8.0.1
[TRT]    loading NVIDIA plugins...
[TRT]    Registered plugin creator - ::GridAnchor_TRT version 1
[TRT]    Registered plugin creator - ::GridAnchorRect_TRT version 1
[TRT]    Registered plugin creator - ::NMS_TRT version 1
[TRT]    Registered plugin creator - ::Reorg_TRT version 1
[TRT]    Registered plugin creator - ::Region_TRT version 1
[TRT]    Registered plugin creator - ::Clip_TRT version 1
[TRT]    Registered plugin creator - ::LReLU_TRT version 1
[TRT]    Registered plugin creator - ::PriorBox_TRT version 1
[TRT]    Registered plugin creator - ::Normalize_TRT version 1
[TRT]    Registered plugin creator - ::ScatterND version 1
[TRT]    Registered plugin creator - ::RPROI_TRT version 1
[TRT]    Registered plugin creator - ::BatchedNMS_TRT version 1
[TRT]    Registered plugin creator - ::BatchedNMSDynamic_TRT version 1
[TRT]    Could not register plugin creator -  ::FlattenConcat_TRT version 1
[TRT]    Registered plugin creator - ::CropAndResize version 1
[TRT]    Registered plugin creator - ::DetectionLayer_TRT version 1
[TRT]    Registered plugin creator - ::EfficientNMS_ONNX_TRT version 1
[TRT]    Registered plugin creator - ::EfficientNMS_TRT version 1
[TRT]    Registered plugin creator - ::Proposal version 1
[TRT]    Registered plugin creator - ::ProposalLayer_TRT version 1
[TRT]    Registered plugin creator - ::PyramidROIAlign_TRT version 1
[TRT]    Registered plugin creator - ::ResizeNearest_TRT version 1
[TRT]    Registered plugin creator - ::Split version 1
[TRT]    Registered plugin creator - ::SpecialSlice_TRT version 1
[TRT]    Registered plugin creator - ::InstanceNormalization_TRT version 1
[TRT]    detected model format - ONNX  (extension '.onnx')
[TRT]    desired precision specified for GPU: FASTEST
[TRT]    requested fasted precision for device GPU without providing valid calibrator, disabling INT8
[TRT]    [MemUsageChange] Init CUDA: CPU +197, GPU +0, now: CPU 225, GPU 2819 (MiB)
[TRT]    native precisions detected for GPU:  FP32, FP16
[TRT]    selecting fastest native precision for GPU:  FP16
[TRT]    attempting to open engine cache file models/iha/best.onnx.1.1.8001.GPU.FP16.engine
[TRT]    loading network plan from engine cache... models/iha/best.onnx.1.1.8001.GPU.FP16.engine
[TRT]    device GPU, loaded models/iha/best.onnx
[TRT]    [MemUsageChange] Init CUDA: CPU +0, GPU +0, now: CPU 249, GPU 2842 (MiB)
[TRT]    Loaded engine size: 23 MB
[TRT]    [MemUsageSnapshot] deserializeCudaEngine begin: CPU 249 MiB, GPU 2842 MiB
[TRT]    Using cublas a tactic source
[TRT]    [MemUsageChange] Init cuBLAS/cuBLASLt: CPU +157, GPU +159, now: CPU 413, GPU 3007 (MiB)
[TRT]    Using cuDNN as a tactic source
[TRT]    [MemUsageChange] Init cuDNN: CPU +241, GPU +244, now: CPU 654, GPU 3251 (MiB)
[TRT]    [MemUsageChange] Init cuBLAS/cuBLASLt: CPU +0, GPU +0, now: CPU 654, GPU 3251 (MiB)
[TRT]    Deserialization required 2598739 microseconds.
[TRT]    [MemUsageSnapshot] deserializeCudaEngine end: CPU 654 MiB, GPU 3251 MiB
[TRT]    [MemUsageSnapshot] ExecutionContext creation begin: CPU 654 MiB, GPU 3251 MiB
[TRT]    Using cublas a tactic source
[TRT]    [MemUsageChange] Init cuBLAS/cuBLASLt: CPU +0, GPU +0, now: CPU 654, GPU 3251 (MiB)
[TRT]    Using cuDNN as a tactic source
[TRT]    [MemUsageChange] Init cuDNN: CPU +0, GPU +0, now: CPU 654, GPU 3251 (MiB)
[TRT]    Total per-runner device memory is 17371136
[TRT]    Total per-runner host memory is 177408
[TRT]    Allocated activation device memory of size 18637824
[TRT]    [MemUsageSnapshot] ExecutionContext creation end: CPU 654 MiB, GPU 3251 MiB
[TRT]
[TRT]    CUDA engine context initialized on device GPU:
[TRT]       -- layers       200
[TRT]       -- maxBatchSize 1
[TRT]       -- deviceMemory 18637824
[TRT]       -- bindings     5
[TRT]       binding 0
                -- index   0
                -- name    'images'
                -- type    FP32
                -- in/out  INPUT
                -- # dims  4
                -- dim #0  1
                -- dim #1  3
                -- dim #2  640
                -- dim #3  640
[TRT]       binding 1
                -- index   1
                -- name    '438'
                -- type    FP32
                -- in/out  OUTPUT
                -- # dims  5
                -- dim #0  1
                -- dim #1  3
                -- dim #2  80
                -- dim #3  80
                -- dim #4  6
[TRT]       binding 2
                -- index   2
                -- name    '490'
                -- type    FP32
                -- in/out  OUTPUT
                -- # dims  5
                -- dim #0  1
                -- dim #1  3
                -- dim #2  40
                -- dim #3  40
                -- dim #4  6
[TRT]       binding 3
                -- index   3
                -- name    '542'
                -- type    FP32
                -- in/out  OUTPUT
                -- # dims  5
                -- dim #0  1
                -- dim #1  3
                -- dim #2  20
                -- dim #3  20
                -- dim #4  6
[TRT]       binding 4
                -- index   4
                -- name    'output'
                -- type    FP32
                -- in/out  OUTPUT
                -- # dims  3
                -- dim #0  1
                -- dim #1  25200
                -- dim #2  6
[TRT]
[TRT]    binding to input 0 images  binding index:  0
[TRT]    binding to input 0 images  dims (b=1 c=3 h=640 w=640) size=4915200
[TRT]    binding to output 0 output  binding index:  4
[TRT]    binding to output 0 output  dims (b=1 c=25200 h=6 w=1) size=604800
[TRT]    binding to output 1 images  binding index:  0
[TRT]    binding to output 1 images  dims (b=1 c=3 h=640 w=640) size=4915200
[TRT]
[TRT]    device GPU, models/iha/best.onnx initialized.
[TRT]    detectNet -- number object classes:  6
[TRT]    detectNet -- maximum bounding boxes:  25200
[TRT]    detectNet -- loaded 6 class info entries
[TRT]    detectNet -- number of object classes:  6
[gstreamer] opening gstCamera for streaming, transitioning pipeline to GST_STATE_PLAYING
[gstreamer] gstreamer changed state from NULL to READY ==> mysink
[gstreamer] gstreamer changed state from NULL to READY ==> capsfilter1
[gstreamer] gstreamer changed state from NULL to READY ==> nvvconv0
[gstreamer] gstreamer changed state from NULL to READY ==> capsfilter0
[gstreamer] gstreamer changed state from NULL to READY ==> nvarguscamerasrc0
[gstreamer] gstreamer changed state from NULL to READY ==> pipeline0
[gstreamer] gstreamer changed state from READY to PAUSED ==> capsfilter1
[gstreamer] gstreamer changed state from READY to PAUSED ==> nvvconv0
[gstreamer] gstreamer changed state from READY to PAUSED ==> capsfilter0
[gstreamer] gstreamer stream status CREATE ==> src
[gstreamer] gstreamer changed state from READY to PAUSED ==> nvarguscamerasrc0
[gstreamer] gstreamer changed state from READY to PAUSED ==> pipeline0
[gstreamer] gstreamer stream status ENTER ==> src
[gstreamer] gstreamer message new-clock ==> pipeline0
[gstreamer] gstreamer changed state from PAUSED to PLAYING ==> capsfilter1
[gstreamer] gstreamer changed state from PAUSED to PLAYING ==> nvvconv0
[gstreamer] gstreamer changed state from PAUSED to PLAYING ==> capsfilter0
[gstreamer] gstreamer changed state from PAUSED to PLAYING ==> nvarguscamerasrc0
[gstreamer] gstreamer message stream-start ==> pipeline0
GST_ARGUS: Creating output stream
CONSUMER: Waiting until producer is connected...
GST_ARGUS: Available Sensor modes :
GST_ARGUS: 3264 x 2464 FR = 21.000000 fps Duration = 47619048 ; Analog Gain range min 1.000000, max 10.625000; Exposure Range min 13000, max 683709000;

GST_ARGUS: 3264 x 1848 FR = 28.000001 fps Duration = 35714284 ; Analog Gain range min 1.000000, max 10.625000; Exposure Range min 13000, max 683709000;

GST_ARGUS: 1920 x 1080 FR = 29.999999 fps Duration = 33333334 ; Analog Gain range min 1.000000, max 10.625000; Exposure Range min 13000, max 683709000;

GST_ARGUS: 1640 x 1232 FR = 29.999999 fps Duration = 33333334 ; Analog Gain range min 1.000000, max 10.625000; Exposure Range min 13000, max 683709000;

GST_ARGUS: 1280 x 720 FR = 59.999999 fps Duration = 16666667 ; Analog Gain range min 1.000000, max 10.625000; Exposure Range min 13000, max 683709000;

GST_ARGUS: 1280 x 720 FR = 120.000005 fps Duration = 8333333 ; Analog Gain range min 1.000000, max 10.625000; Exposure Range min 13000, max 683709000;

GST_ARGUS: Running with following settings:
   Camera index = 0
   Camera mode  = 5
   Output Stream W = 1280 H = 720
   seconds to Run    = 0
   Frame Rate = 120.000005
GST_ARGUS: Setup Complete, Starting captures for 0 seconds
GST_ARGUS: Starting repeat capture requests.
CONSUMER: Producer has connected; continuing.
[gstreamer] gstCamera -- onPreroll
[gstreamer] gstBufferManager recieve caps:  video/x-raw(memory:NVMM), width=(int)1280, height=(int)720, format=(string)NV12, framerate=(fraction)30/1
[gstreamer] gstBufferManager -- recieved first frame, codec=raw format=nv12 width=1280 height=720 size=1008
[gstreamer] gstBufferManager -- recieved NVMM memory
[gstreamer] gstreamer changed state from READY to PAUSED ==> mysink
[gstreamer] gstreamer message async-done ==> pipeline0
[gstreamer] gstreamer changed state from PAUSED to PLAYING ==> mysink
[gstreamer] gstreamer changed state from PAUSED to PLAYING ==> pipeline0
RingBuffer -- allocated 4 buffers (2764800 bytes each, 11059200 bytes total)
[TRT]    1: [executionContext.cpp::executeInternal::652] Error Code 1: Cuda Runtime (unspecified launch failure)
[TRT]    failed to execute TensorRT context on device GPU

[TRT]    ------------------------------------------------
[TRT]    Timing Report models/iha/best.onnx
[TRT]    ------------------------------------------------
[cuda]      unspecified launch failure (error 719) (hex 0x2CF)
[cuda]      /home/nano/jetson-inference/build/aarch64/include/jetson-inference/tensorNet.h:698
[TRT]    Pre-Process   CPU   0.06578ms  CUDA   0.00000ms
[cuda]      unspecified launch failure (error 719) (hex 0x2CF)
[cuda]      /home/nano/jetson-inference/build/aarch64/include/jetson-inference/tensorNet.h:698
[TRT]    Network       CPU   0.00000ms  CUDA   0.00000ms
[TRT]    Total         CPU   0.06578ms  CUDA   0.00000ms
[TRT]    ------------------------------------------------

[TRT]    note -- when processing a single image, run 'sudo jetson_clocks' before
                to disable DVFS for more accurate profiling/timing measurements

[cuda]      unspecified launch failure (error 719) (hex 0x2CF)
[cuda]      /home/nano/jetson-inference/utils/codec/gstBufferManager.cpp:288
[gstreamer] gstDecoder -- failed to retrieve next image buffer
detectnet:  failed to capture video frame
[cuda]      unspecified launch failure (error 719) (hex 0x2CF)
[cuda]      /home/nano/jetson-inference/utils/codec/gstBufferManager.cpp:288
[gstreamer] gstDecoder -- failed to retrieve next image buffer
detectnet:  failed to capture video frame
....
....
....

Where am I doing wrong above? The ./detectnet command works fine? I would be glad if you help. Good work.

**Comment by dusty-nv** [comment]
@muttakigenc sorry for the delay - if this is still an issue, please check that your MIPI CSI camera works first with the `nvgstcapture-1.0` tool.

**Comment by muttakigenc** [comment]
First of all, thanks. best.onnx is a yolov5 model file. I guess this is the cause of the error. Because it works with other models. So it has nothing to do with the camera connection. To solve this problem, I did the training I did on yolo here again with pytorch.

This shouldn't normally be a problem, right?

**Comment by muttakigenc** [comment]
> @muttakigenc sorry for the delay - if this is still an issue, please check that your MIPI CSI camera works first with the `nvgstcapture-1.0` tool.

First of all, thanks. best.onnx is a yolov5 model file. I guess this is the cause of the error. Because it works with other models. So it has nothing to do with the camera connection. To solve this problem, I did the training I did on yolo here again with pytorch.

This shouldn't normally be a problem, right?

**Comment by dusty-nv** [comment]
> This shouldn't normally be a problem, right?

Oh right, no it shouldn't be - your errors were most likely caused by invalid memory accesses due to the fact that my detection post-processing is setup for SSD-Mobilenet, not YOLO.
